Added support for displaying time taken by each player & also Blitz mode Chess

kiran9k opened the pull request:
The time played by each player is being displayed along with the user name . Also , one can set a timeout ,  inorder to play in Blitz mode (to be specified in chesshub.js file)

benas commented:
Cool! Thank you for this addition :+1: 

But I think your fork is not up to date with last commits in the master branch.
I have added some comments on outdated code parts. Please make sure to update your branch.

I'll merge your pull request asap.
Thank you for your contribution :-)
Cheers

benas commented on .gitignore:
```diff
@@ -1,31 +0,0 @@
-# Logs
```
can you undo the deletion of the .gitignore file? may be deleted by error?

benas commented on views/layout.hbs:
```diff
@@ -1,5 +1,5 @@
 <!DOCTYPE html>
-<html>
+<html ng-app="chesshubApp">
```
This is outdated comparing to master

benas commented on views/layout.hbs:
```diff
@@ -40,7 +40,7 @@
     <div class="footer">
         <p>&copy; ChessHub 2014, <i class="fa fa-code"></i> with <i class="fa fa-heart"></i> by
-            <a target="_blank" href="http://www.mahmoud-benhassine.fr">Mahmoud Ben Hassine</a></p>
+            <a target="_blank" href="http://www.mahmoudbenhassine.com">Mahmoud Ben Hassine</a></p>
```
This is outdated comparing to master
(Should be mahmoud-benhassine.fr, see 7f4b96a4641c8f01c67916c57cb7e0cde5c2bde9)

benas commented on public/javascripts/chesshub.js:
```diff
@@ -133,6 +134,48 @@ $( document ).ready(function() {
         var opponentSide = side === "black" ? "white" : "black";
 
+
+        /*
```
This is the heart of the pull request. Great! don't touch it :-)

benas commented on app.js:
```diff
@@ -47,6 +47,12 @@ var login = require('./routes/login');
+app.use(function(req, res, next) {
+    res.header("Access-Control-Allow-Origin", "*");
+    res.header("Access-Control-Allow-Headers", "Origin, X-Requested-With, Content-Type, Accept");
+    next();
```
Please send this change in a separate pull request.
I have created an issue for that. see here: #8

benas commented on app.js:
```diff
@@ -54,13 +60,24 @@ app.use('/account', account);
 
+
+
+var url_path='192.168.12.77';
```
Host and port configuration is not related to adding time taken by each player.
Please send this change in a separate pull request.
I have created an issue for that. see here: #7

benas commented on public/javascripts/chesshub.js:
```diff
@@ -205,7 +250,7 @@ $( document ).ready(function() {
          */
         socket.on('wait', function () {
-            var url = "http:/localhost:3000/game/" + token + "/" + opponentSide;
+            var url = "http:/192.168.12.77:3000/game/" + token + "/" + opponentSide;
```
Should be fixed in issue #7

benas commented on public/javascripts/chesshub.js:
```diff
@@ -364,7 +414,7 @@ $( document ).ready(function() {
             $.ajax({
                 type: "POST",
-                url: "http://localhost:3000/search",
+                url: "http://192.168.12.77:3000/search",
```
Should be fixed in issue #7

benas commented on views/partials/game.hbs:
```diff
@@ -9,13 +9,15 @@
+                <p>
                     <img src="/images/chesspieces/wikipedia/bK.png" class="side-img" alt="Black"/>
                     <span id="player-black"></span> <span class="turn" id="turn-b">&nbsp;<i class=""></i></span>
+                    <span id="timeb" >{{time}}</span>
```
That's fine, to keep as is.

benas commented on public/javascripts/chesshub.js:
```diff
@@ -12,13 +13,13 @@ $( document ).ready(function() {
 
     // socket used for real time games
-    var socket = io('http://localhost:3000', { query: 'user=' + username });
+    var socket = io('http://192.168.12.77:3000', { query: 'user=' + username });
```
Should be fixed in issue #7

benas commented on public/javascripts/chesshub.js:
```diff
@@ -12,13 +13,13 @@ $( document ).ready(function() {
 
     //socket used to broadcast live games on tv page
-    var tvSocket = io('http://localhost:3000/tv');
+    var tvSocket = io('http://192.168.12.77:3000/tv');
```
Should be fixed in issue #7

benas commented on public/javascripts/chesshub.js:
```diff
@@ -12,13 +13,13 @@ $( document ).ready(function() {
 
     // socket used to broadcast events to monitoring page
-    var monitorSocket = io('http://localhost:3000/monitor');
+    var monitorSocket = io('http://192.168.12.77:3000/monitor');
```
Should be fixed in issue #7

benas commented:
A side note, you added handlebars-v2.0.0.js but you didn't imported it in the project. 
I'm not sure this is needed since handlebars is already imported:

https://github.com/benas/gamehub.io/blob/master/package.json#L19
https://github.com/benas/gamehub.io/blob/master/app.js#L28

That's all for my review.
Do not hesitate to add your comments. 
If it is ok for you, please update the pull request and I'll merge it asap. 
Your contribution is welcome!

kiran9k commented:
Hi ,
I just updated the fork to the latest . 
I have added codes to display the time played by each player along with the user name . Also , support to Blitz mode chess is also provided .

Cheers

benas commented:
Cool! this goes straight to the master branch :-)

Thank you for your contribution! :+1: 

Regards
Mahmoud
